.github/actions/deploy/action.yml

```diff
@@ -61,59 +72,88 @@ runs:
             --slot ${{ inputs.slot_name }} `
+            --query principalId `
+            --output tsv
+          Write-Host '✅ identity created'
+
+          # Configure WebApp to use the Manage Identity Credentials to perform docker pull operations
+          $configId = (
+            az webapp show `
+              --resource-group ${{ inputs.AZURE_RESOURCE_GROUP }} `
+              --name ${{ inputs.APP_SERVICE_NAME }} `
+              --slot ${{ inputs.slot_name }} `
+              --query id `
+              --output tsv
+            ) + "/config/web"
+          az resource update `
+            --ids $configId `
+            --set properties.acrUseManagedIdentityCreds=True `
+            --output none
+          Write-Host '✅ acrUseManagedIdentityCreds set'
+
+          $acrId = az acr show `
+            --resource-group ${{ inputs.AZURE_RESOURCE_GROUP }} `
+            --name ${{ inputs.ACR_LOGIN_SERVER }} `
             --query id `
             --output tsv
-          ) + "/config/web"
-        az resource update `
-          --ids $configId `
-          --set properties.acrUseManagedIdentityCreds=True `
-          --output none
-        Write-Host '✅ acrUseManagedIdentityCreds set'
-
-        $acrId = az acr show `
-          --resource-group ${{ inputs.AZURE_RESOURCE_GROUP }} `
-          --name ${{ inputs.ACR_LOGIN_SERVER }} `
-          --query id `
-          --output tsv
-        Write-Host '✅ acr found'
+          Write-Host '✅ acr found'
+
+          # grant access to the identity on ACR
+          az role assignment create `
+            --assignee-object-id $identityId `
+            --assignee-principal-type ServicePrincipal `
+            --scope $acrId `
+            --role acrpull `
+            -o none
+          Write-Host '✅ acrpull role granted'
+
+          $kvId = az keyvault show `
+            --resource-group ${{ inputs.AZURE_RESOURCE_GROUP }} `
+            --name ${{ inputs.KEY_VAULT }} `
+            --query id `
+            --output tsv
+
+          if($kvId) {
+          Write-Host '✅ KeyVault found'
+
+          # grant access to the identity on KeyVault
+          az role assignment create `
+            --assignee-object-id $identityId `
+            --assignee-principal-type ServicePrincipal `
+            --scope $kvId `
+            --role "Key Vault Secrets User" `
+            -o none
+          Write-Host '✅ Key Vault Secrets User role granted'
+          } else {
+            Write-Host '❌ Key Vault not found'
+          }
+        }
+        # Apply app settings to the PR slots
+        az webapp config appsettings set `
+          --resource-group ${{ env.AZURE_RESOURCE_GROUP }} `
+          --name ${{ inputs.APP_SERVICE_NAME }} `
+          --slot ${{ inputs.slot_name }} `
+          --settings "GOOGLE_RECAPTCHA_KEY_V2=${{ inputs.GOOGLE_RECAPTCHA_KEY_V2 }}" "GOOGLE_RECAPTCHA_SITE_KEY=${{ inputs.GOOGLE_RECAPTCHA_SITE_KEY }}" "CREATE_LEAD_ENDPOINT=${{ inputs.CREATE_LEAD_ENDPOINT }}"
```

Comment:
for now - can we guard this so that it is only applied on pr slots?

Reply:
Isn't it already guarded with the GitHub action name "pull_request"?